src/pyfmi/fmi.pyx

```diff
@@ -496,33 +513,49 @@ cdef class ModelBase:
         for i in range(N):
             print(log[i])
 
-    def extract_xml_log(self, file_name=None):
+    def extract_xml_log(self, file_name=None, stream = None):
```

Comment:
Why do we need both? Should we not just have the functionality to extract the log to a user-defined place, be it either a file or a stream. So can we not just have one argument?

Reply:
I added the two arguments because the user can with argument 'stream' now extract_xml_log either from a separate stream or from the the internal stream from the loading of the fmu, i.e.:
```
stream1 = ...
stream2 = ... # some empty stream
stream3 = ...
fmu = load_fmu(compiled_fmu, log_filename = stream1)
fmu.simulate(a, b)
fmu = load_fmu(compiled_fmu, log_filename = stream3)
fmu.simulate(a, c)
fmu.extract_xml_log() # will extract from stream3 and write to a file as default
fmu.extract_xml_log(file_name = stream2) # will extract from stream3 but extract it into stream2 instead of generating a file.
fmu.extract_xml_log(stream = stream1) # will extract from stream1 and generate a file
```

Reply:
I see, if the user wants to extract the xml from a separate stream, they should use directly the "extract_xml_log" from the common package. I don't think we should support it here. I.e. the last case should not be supported.

Reply:
Fixed with the comment above (where I linked to this changeset https://github.com/modelon-community/PyFMI/commit/8e355093157fb6e9b1824156bc923aa8a873cb86)